### PR TITLE
add endpoint to edit repository

### PR DESCRIPTION
**Pull request by eunomie**
### Proposed changes

Describe endpoint to update repository's settings:

- description
- full description (readme)
- public/private

### CODE REVIEW

File: docker-hub/api/latest.yaml [hunk 1]
```diff
@@ -1,13 +1,7 @@
-swagger: '2.0'
```
Comment (technicallyjosh) [inline docker-hub/api/latest.yaml:1]: Doing a major upgrade here breaks our current generation of some APIs and what is copied into here. So each time a change occurs in an API we then have to translate from swagger 2 to 3.

**Review by technicallyjosh — CHANGES_REQUESTED** [review]
I'd request that we keep swagger 2 until we can get all our internal API docs (generated) into OpenAPI 3. Have you also looked at the effect you have on the Engine docs? They use Redoc as well for that generation and last time I tried upgrading Redoc, it broke lots of things over there.

**Comment by technicallyjosh** [comment]
It would also be helpful if we kept major upgrades as separate PRs 😄

**Comment by eunomie** [comment]
OK understood, I'll go back with swagger and a small hack to have it working in all cases.

File: docker-hub/api/latest.yaml [hunk 2]
```diff
@@ -94,6 +94,10 @@ tags:
     x-displayName: Org Settings
     description: |
       The Org Settings API endpoints allow you to manage your organization's settings.
+  - name: repository
```
Comment (technicallyjosh) [inline docker-hub/api/latest.yaml:97]: **question:** Shouldn't we call this "Repositories" (plural)? Will make it easier to add onto with more routes.

**Review by technicallyjosh — APPROVED** [review]
LGTM. Just had a question about the naming 😄

File: docker-hub/api/latest.yaml (hunk 2 above)
Comment (eunomie) [reply docker-hub/api/latest.yaml:97]: 👍 I'll update it

**Review by technicallyjosh — APPROVED** [review]
LGTM! Thanks!

**Review by usha-mandya — APPROVED** [review]
LGTM. Thank you @eunomie

**Pull request merged**